src/components/NewSnippet.jsx

```diff
@@ -20,28 +20,35 @@ class NewSnippet extends React.Component {
       tags: [],
       syntax: '', // eslint-disable-line react/no-unused-state
     };
+    this.onKeyPress = (e) => {
+      if (e.which === 13) { e.preventDefault(); }
+    };
     this.postSnippet = this.postSnippet.bind(this);
     this.onSyntaxClick = this.onSyntaxClick.bind(this);
     this.onInputChange = this.onInputChange.bind(this);
+    this.onTagAdded = this.onTagAdded.bind(this);
+    this.onTagRemoved = this.onTagRemoved.bind(this);
   }
 
   componentDidMount() {
     const { dispatch } = this.props;
-
     dispatch(actions.fetchSyntaxes);
   }
 
+  onTagAdded(tag) {
+    this.setState({ tags: [...this.state.tags, tag] });
```

Comment:
nit: I think we currently do not enforce it at the API level, but maybe we should prevent duplicates of tags on the same snippet? @ikalnytskyi what do you think?

Reply:
I think it makes sense. Will you file an issue for API to enforce that?

Reply:
@ikalnytskyi do you want to reject requests with duplicates with `400 Bad Request`? Or just remove the duplicates before actually saving the snippet to the db?

Reply:
@malor Honestly, I don't have strong opinion here. Both approaches have their own pros and cons. What do you think?